build-support/ci.py

```diff
@@ -32,46 +33,62 @@ def __str__(self):
 
 def main() -> None:
   args = create_parser().parse_args()
-  with setup_pants_version(args.pants_version):
-    with setup_python_version(args.python_version):
-      run_tests(skip_pantsd_tests=args.skip_pantsd_tests)
+  envs = itertools.product(args.pants_versions, args.python_versions)
+  for pants_version, python_version in envs:
+    run_tests_with_env(
+      pants_version=pants_version,
+      python_version=python_version,
+      skip_pantsd_tests=args.skip_pantsd_tests
+    )
 
 
 def create_parser() -> argparse.ArgumentParser:
   parser = argparse.ArgumentParser(description="Utility to run CI for the setup repo.")
   parser.add_argument(
-      "--pants-version",
+      "--pants-versions",
       action="store",
       type=PantsVersion,
       choices=list(PantsVersion),
       required=True,
+      nargs="+",
       help="Pants version to configure ./pants to use."
   )
   parser.add_argument(
-      "--python-version",
+      "--python-versions",
       action="store",
       type=PythonVersion,
       choices=list(PythonVersion),
       required=True,
+      nargs="+",
       help="Python version to configure ./pants to use."
   )
   parser.add_argument("--skip-pantsd-tests", action="store_true")
   return parser
 
 
+def run_tests_with_env(*,
+    pants_version: PantsVersion, python_version: PythonVersion, skip_pantsd_tests: bool
+  ) -> None:
+  slug = f"Tests_{pants_version}_{python_version}"
+  banner_message = f"Running tests with `--pants-version={pants_version}` and `--python_version={python_version}`."
+  with travis_section(slug, banner_message):
+    with setup_pants_version(pants_version), setup_python_version(python_version):
+      run_tests(skip_pantsd_tests=skip_pantsd_tests)
```

Comment:
In the pants repo, I typically see this done as:
```python
 with travis_section(slug, banner_message),\
      setup_pants_version(pants_version),\
      setup_python_version(python_version):
    run_tests(skip_pantsd_tests=skip_pantsd_tests)
```
and the vertical alignment feels easier for me to read, personally. But using multiple `with`s is definitely also useful to demarcate logically separate contexts.